Functions/InputOutput/xASL_io_dcm2nii.m

```diff
@@ -349,14 +349,14 @@
                 % Make BIDS compatible dest_file here
                 DestFileName = series_name;
 
-                [Ind1, Ind2] = regexp(niiNames{iVolume},'_run-\d*_','start','end');
-                if ~isempty(Ind1) && ~isempty(Ind2) % add run_index suffix (if it is there)
-                    RunName = niiNames{iVolume}(Ind1+1:Ind2-1);
-                    DestFileName = [DestFileName '_' RunName];
-                end
-                if ~isempty(ContrastAdd{iVolume}) % add contrast suffix (if there is something that indicates it)
-                    DestFileName = [DestFileName '_' ContrastAdd{iVolume}];
-                end
+                %[Ind1, Ind2] = regexp(niiNames{iVolume},'_run-\d*_','start','end');

```

Comment:
Regarding old edits/commits - as I stated from the start: the directory (ASL_1, ASL_2) is the correct identifier for the Runs that is used in your original DICOM2NII and then also NII2BIDS scheme. No other identifier for Runs is not needed because there is supposed to be only a single Run per modality in the directory of that Run. 

So all your edits were useless because they were solving a case that should have never happened. Sorry, but I told you so from the start ;)

The real reason for the error was that we were (falsely, I have no idea how that code got there) checking DICOM.ProtocolDescription tag and extracting "run-\d*" from there. That's something that will never work - we can never reliably trust ProtocolDescription tag. Luckily, that code was always ignored until we got the data from Marcus that named (by coincidence) the Sequence "ASL_PCASL_HAD4_blahblah_run_02".

This is the case irrespective of dcm2niix version.

I will still re-run it on all flavors, but I can't really find a reason why we had this line of code there.

Reply:
The code that Bea and I created just had a more flexible approach to finding NIfTI files, without any change in the identifier. Check it out!

Reply:
Conclusion - we keep the current version. We will do the change that "run_1" is always added in #1180.

Reply:
Now the warnings are similar to as they were in your commits.